core/src/main/java/org/fao/geonet/util/LimitedInputStream.java

```
@@ -0,0 +1,165 @@
+package org.fao.geonet.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * Like a {@link org.apache.commons.io.input.BoundedInputStream} but includes a method to check if the limit has been reached.
+ */
+public class LimitedInputStream extends InputStream {
+    private final InputStream in;
+    private final long max;
+    private long pos;
+    private long mark;
+    private boolean propagateClose;
+    private boolean limitReached;
+
+    /**
+     * Creates a new instance.
+     *
+     * @param in   the input stream
+     * @param size the maximum number of bytes to read
+     */
+    public LimitedInputStream(InputStream in, long size) {
+        this.pos = 0L;
+        this.mark = -1L;
+        this.propagateClose = true;
+        this.max = size;
+        this.in = in;
+    }
+
+    /**
+     * Creates a new instance.
+     *
+     * @param in the input stream
+     */
+    public LimitedInputStream(InputStream in) {
+        this(in, -1L);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * <p>In this implementation if the limit has been reached, returns -1.</p>
+     */
+    @Override
+    public int read() throws IOException {
+        if (this.max >= 0L && this.pos >= this.max) {
+            this.limitReached = true;
+            return -1;
```

I would expect that this class should throw an exception based on IOException 

```
public class LimitExceededException extends IOException {
    public LimitExceededException(String message) {
        super(message);
    }
}
```

And then instead of returning -1 it would do something like the following.
          `  throw new LimitExceededException(....)`

---

Fixed in the latest push. I am instead extending the abstract class org.apache.commons.fileupload.util.LimitedInputStream which only needs an implementation of the raiseError method in which I throw a new InputStreamLimitExceededException which extends IOException.